tracer/src/Datadog.Trace/Iast/InsecureAuthAnalyzer.cs

```diff
@@ -0,0 +1,115 @@
+// <copyright file="InsecureAuthAnalyzer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Linq;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+#if !NETFRAMEWORK
+using Microsoft.AspNetCore.Http;
+using Microsoft.Net.Http.Headers;
+#endif
+
+namespace Datadog.Trace.Iast;
+
+#nullable enable
+
+internal static class InsecureAuthAnalyzer
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(InsecureAuthAnalyzer));
+
+#if !NETFRAMEWORK
+    public static void AnalyzeInsecureAuth(IHeaderDictionary headers, IntegrationId integrationId, int statusCode)
+    {
+        try
+        {
+            // Check for Authorization header
+            const string headerName = HeaderNames.Authorization;
+            if (!headers.TryGetValue(headerName, out var authHeader))
+            {
+                return;
+            }
+
+            if (IgnoredStatusCode(statusCode))
+            {
+                return;
+            }
+
+            var detectedScheme = InsecureSchemeDetected(authHeader);
```

Comment:
I wouldn't handle the `StringValues` like this. By converting it to a `string`, you're forcing extra allocations, and then you're splitting it again later which is expensive. instead, I suggest your handle both cases explicitly, e.g. something like this:

```csharp
if(string.IsNullOrEmpty(authHeader))
{
    return
}

if(authHeader.Count == 0)
{
  // safe to cast to string
  InsecureSchemeDetected(authHeader);
} 
else
{
    // we know it's an array
    foreach(var header in authHeader)
    {
        InsecureSchemeDetected(authHeader);
    }
}
```

Reply:
fixed in b1e18bd